pkg/trace/obfuscate/sql.go

```diff
@@ -44,21 +44,25 @@ type tokenFilter interface {
 
 // discardFilter is a token filter which discards certain elements from a query, such as
 // comments and AS aliases by returning a nil buffer.
-type discardFilter struct{}
+type discardFilter struct {
+	newSQLNormalization bool
+}
 
 // Filter the given token so that a `nil` slice is returned if the token is in the token filtered list.
 func (f *discardFilter) Filter(token, lastToken TokenKind, buffer []byte) (TokenKind, []byte, error) {
 	// filters based on previous token
 	switch lastToken {
 	case FilteredBracketedIdentifier:
 		if token != ']' {
-			// we haven't found the closing bracket yet, keep going
-			if token != ID {
-				// the token between the brackets *must* be an identifier,
+			switch token {
+			case ID, '.':
+				// we haven't found the closing bracket yet, keep going
```

Comment:
This comment is addressing the `if` statement above. Here it might not make sense. Can you please put it back?

Reply:
Please explain why `'.'` is an identifier, in the replacement comment for this one.